codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/TelemetryGenerator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.generators;
+
+import java.util.Comparator;
+import java.util.Set;
+import java.util.logging.Logger;
+import software.amazon.smithy.codegen.core.directed.ContextualDirective;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.ruby.codegen.GenerationContext;
+import software.amazon.smithy.ruby.codegen.RubyCodeWriter;
+import software.amazon.smithy.ruby.codegen.RubyFormatter;
+import software.amazon.smithy.ruby.codegen.RubySettings;
+import software.amazon.smithy.ruby.codegen.util.Streaming;
+import software.amazon.smithy.utils.StringUtils;
+
+public class TelemetryGenerator extends RubyGeneratorBase {
+    private static final Logger LOGGER =
+            Logger.getLogger(ParamsGenerator.class.getName());
+
+    private final Set<OperationShape> operations;
+
+    public TelemetryGenerator(ContextualDirective<GenerationContext, RubySettings> directive) {
+        super(directive);
+        this.operations = directive.operations();
+    }
+
+    @Override
+    protected String getModule() {
+        return "Telemetry";
+    }
+
+    public void render() {
+        write(writer -> {
+            writer
+                    .preamble()
+                    .includeRequires()
+                    .addModule(settings.getModule())
+                    .apiPrivate()
+                    .addModule("Telemetry")
+                    .call(() -> renderOperations(writer))
+                    .write("")
+                    .closeAllModules();
+        });
+        LOGGER.fine("Wrote telemetry to " + rbFile());
+    }
+
+    private void renderOperations(RubyCodeWriter writer) {
+        operations.stream()
+                .filter((o) -> !Streaming.isEventStreaming(model, o))
```

Comment:
Totally out of scope for this PR, but we'll need to figure out how telemetry should work for event stream operations....